src/java.base/share/classes/java/util/function/UnaryOperator.java

```diff
@@ -49,4 +51,46 @@ public interface UnaryOperator<T> extends Function<T, T> {
     static <T> UnaryOperator<T> identity() {
         return t -> t;
     }
+
+    /**
+     * Returns a composed function that first applies this unary operator to
+     * its input, and then applies the {@code after} unary function to the result.
+     * If evaluation of either function throws an exception, it is relayed to
+     * the caller of the composed function.
+     *
+     * @param after the unary operator to apply after this unary operator is applied
+     * @return a composed unary operator that first applies this function and then
+     * applies the {@code after} unary operator
+     * @throws NullPointerException if after is null
+     *
+     * @see Function#andThen(Function)
+     */
+    default UnaryOperator<T> andThenUnary(UnaryOperator<T> after) {
```

Comment:
What are your comments on this proposed method?

Reply:
Just curious, do we have any idea how extensively `UnaryOperator` is used as a parameter type in user code compared to plain `Function<T, T>`? If it's not preferred over `Function<T, T>`, this new method might not be very useful.

Reply:
Generally speaking, there are 1,594 usages of `Function` and 208 usages of `UnaryFunction` in the JDK itself. More specifically, there are 8 usages of `Function::andThen` and zero use of `UnaryOperator::andThen` in the JDK.

Reply:
I've run some corpus analysis and this came up:

```
pminborg@pminborg-mac open % cat ~/Downloads/experiment-2.log | grep Function.andThen | wc -l
    6689
pminborg@pminborg-mac open % cat ~/Downloads/experiment-2.log | grep UnaryOperator.andThen | wc -l
     153
```